calyx-py/test/correctness/matrix.py

```diff
@@ -0,0 +1,71 @@
+import calyx.builder as cb
+
+def insert_matmul_component(prog):
+    matmul = prog.component("main")
+
+    A = matmul.comb_mem_d2("A", 32, 4, 4, 3, 3, is_external=True)
+    B = matmul.seq_mem_d2("B", 32, 4, 4, 3, 3, is_external=True)
+    C = matmul.seq_mem_d2("C", 32, 4, 4, 3, 3, is_external=True)
+
+    mult = matmul.mult_pipe(32)
+    add = matmul.add(32)
+
+    acc = matmul.reg(32)
+
+    # matrix entries
+    a = matmul.reg(32)
+    b = matmul.reg(32)
+
+    # iterators
+    i = matmul.reg(3)
+    j = matmul.reg(3)
+    k = matmul.reg(3)
+
+    zero_acc = matmul.reg_store(acc, 0)
+    zero_i = matmul.reg_store(i, 0)
+    zero_j = matmul.reg_store(j, 0)
+    zero_k = matmul.reg_store(k, 0)
+
+    cond_i = matmul.lt_use(i.out, 4)
+    cond_j = matmul.lt_use(j.out, 4)
+    cond_k = matmul.lt_use(k.out, 4)
```

Comment:
Now that you are in Python land and not Calyx land, you could consider parameterizing this method by the sizes of the input matrices

Reply:
That's a good idea; I've added this in the latest commit. I'm starting to see how convenient it is to use the eDSL over handwriting calyx